src/ServicePulse.Host/vue/src/components/EventLogItem.vue

```diff
@@ -0,0 +1,186 @@
+<script setup>
+import { useRouter } from "vue-router";
+import TimeSince from "../components/TimeSince.vue";
+
+defineProps({
+  eventLogItem: Object,
+});
+const router = useRouter();
+
+function navigateToEvent(eventLogItem) {
+  switch (eventLogItem.category) {
+    case "Endpoints":
+      router.push("/configuration/endpoint-connection");
+      break;
+    case "HeartbeatMonitoring":
+      window.location = "/a/#/endpoints";
+      break;
+    case "CustomChecks":
+      window.location = "/a/#/custom-checks";
+      break;
+    case "EndpointControl":
+      window.location = "/a/#/endpoints";
+      break;
+    case "MessageFailures":
+      var newlocation = "/failed-messages";
+      if (eventLogItem.related_to && eventLogItem.related_to.length > 0 && eventLogItem.related_to[0].search("message") > 0) {
```

Comment:
first and second clause can be combined to eventLogItem.related_to?.length (don't need to check for >0 since 0 is falsy)

Reply:
Applied